src/module-config/module-config.ts

```diff
@@ -38,7 +19,26 @@ export function provide(config) {
   configs.push(config);
 }
 
-export function getConfig(moduleName) {
+let importMapConfigHasBeenAdded = false;
```

Comment:
This function has a bug when there are multiple calls to `getConfig()` happening **during the time the config is loading**. The `importMapConfigHasBeenAdded` variable will be false for the multiple calls, which will trigger System.import() multiple (which is fine) but then call `configs.unshift(res)` multiple times (which is not fine).

To fix, I suggest an implementation like this:

```js
let configPromise

export function getConfig(moduleName: string): Promise<object> {
  if (configPromise) {
    return configPromise;
  } else {
    return configPromise = Promise.resolve().then(() => {
      let importMapConfigExists;
      try {
        System.resolve("config-file");
        importMapConfigExists = true;
      } catch {
        importMapConfigExists = false;
      }

      if (importMapConfigExists) {
        System.import("config-file").then(res => {
          configs.unshift(res);
          importMapConfigHasBeenAdded = true;
        });
      }
    })
  }
}
```

Reply:
Good catch, and nice solution! Thanks!